retrievalmarket/impl/clientstates/client_states_test.go

```diff
@@ -214,14 +238,16 @@ func TestProcessPaymentRequested(t *testing.T) {
 		fe := environment(testnet.TestDealStreamParams{}, testnodes.TestRetrievalClientNodeParams{
 			Voucher: testVoucher,
 		})
-		f := clientstates.ProcessPaymentRequested(ctx, fe, *dealState)
-		f(dealState)
+		fsmCtx := fsmtest.NewTestContext(ctx, eventMachine)
+		err := clientstates.ProcessPaymentRequested(fsmCtx, fe, *dealState)
+		require.NoError(t, err)
+		fsmCtx.ReplayEvents(t, dealState)
 		require.Empty(t, dealState.Message)
 		require.Equal(t, dealState.PaymentRequested, abi.NewTokenAmount(0))
 		require.Equal(t, dealState.FundsSpent, big.Add(defaultFundsSpent, defaultPaymentRequested))
 		require.Equal(t, dealState.BytesPaidFor, defaultTotalReceived)
 		require.Equal(t, dealState.CurrentInterval, defaultCurrentInterval+defaultIntervalIncrease)
-		require.Equal(t, dealState.Status, retrievalmarket.DealStatusCompleted)
+		require.Equal(t, dealState.Status, retrievalmarket.DealStatusFinalizing)
 	})
 
 	t.Run("not enough funds left", func(t *testing.T) {
```

Comment:
this would really benefit from either table tests or helpers to reduce copy-pasta, or both.